### ISSUE TITLE
On iOS, registering a product after store.refresh() results in that product never loading.

### ISSUE THREAD

**Issue by KoryNunn** [j3k0/cordova-plugin-purchase#1341]
### Observed behavior

On iOS, registering a product after store.refresh() results in that product never loading.

eg:

```
store.register({
  id: productId,
  type: store.NON_CONSUMABLE
});

store.refresh();

store.when(productId, 'loaded', () => { /* never fires*/ });
```

The above works as expected on Android.

Further, `product.loaded` also never changes, so this is not an issue with the `when` handler.

### Expected behavior

The product loads like it does on android.

### System Info

NA

**Comment by fieres** [comment]
From the docs: "Use [store.register()](https://github.com/j3k0/cordova-plugin-purchase/issues/1341#register) before your first call to [store.refresh()](https://github.com/j3k0/cordova-plugin-purchase/issues/1341#refresh)."

So, the observed behavior is the expected behavior. It is undefined what happens if you register products after your first call to refresh()

**Comment by KoryNunn** [comment]
Is there no way to add a product asynchronously? If the product is added while a user has the app open, they can't buy it?

**Comment by fieres** [comment]
I don‘t understand what you mean by „add a product asynchronously“. All products must be created and be approved in Appstore Connect, or Googke Play Console, resp. As far as I know you cannot create products ad hoc in the app. So the set of all available products is always known at app start.

**Comment by j3k0** [comment]
Version of the plugin before version 13 don't support loading more products on iOS (because product definitions where required to process the pending purchases, thus this requirement). Next version is available in beta in PR #1330 and should allow registering more products on all platforms.
